bundle/edu.gemini.model.p1/src/main/scala/edu/gemini/model/p1/immutable/Proposal.scala

```diff
@@ -115,6 +121,13 @@ case class Proposal(meta:Meta,
 
   def resetObservationMeta:Proposal = copy(observations = observations.map(_.copy(meta = None)))
 
+  // REL-3290: Now that we always have an observation (an empty one if there are no others), when we access the
+  // list of observations for processing, we want to filter this dummy observation out.
+  def nonEmptyObservations: List[Observation] = observations.filterNot(_ == Observation.empty)
+
+  // REL-3290: Check to see if there are observations that are non-empty.
+  def hasNonEmptyObservations: Boolean = nonEmptyObservations.nonEmpty
+
```

Comment:
These are just convenience methods for reuse to made sure the empty observation isn't propagated to places it shouldn't be.